docusaurus/docs/iOS/guides/working-with-attachments.md

```diff
@@ -1,3 +1,263 @@
 ---
 title: Working with Attachments/Files
----
\ No newline at end of file
+---
+
+Stream chat allows you to add attachments to a message. There is built-in support for files, images, videos, giphy and URL preview attachments.
+It is also possible to have your own type of attachment and to customize how built-in attachments are rendered. Custom attachments are one of the most useful ways to add application specific content to messages. Examples of custom attachments are: location sharing, workouts and voice memos.
+
+The type of attachment is defined by the value on the `Type` field. All SDKs support these built-in attachment types out-of-the-box:
+
+- attachment type "image" for images
+- attachment type "video" for videos
+- attachment type "giphy" for interactive Giphy attachments (see Slack)
+- attachment type "file" for files
+
+URL previews can have type image, video or audio depending on the resource that the URL addresses. All attachments view classes are subclasses of the `AttachmentViewInjector` class.
+
+All Chat SDKs allow you to customize how built-in types are handled as well as use your own custom attachment types.
+
+## How to customize built-in attachments
+
+Built-in attachments are mapped to specific `AttachmentViewInjector` classes, the mapping is defined on the `Components` class.
+
+**Example:** change the view class used to render files attachments with a custom class.
+
+```swift
+Components.default.filesAttachmentInjector = MyCustomAttachmentViewInjector.self
+```
+
+This is the list of `Components`'s attributes used to map attachments to views
+
+|  Attribute                | Description                                     | Default View Class                              |
+|---------------------------|-------------------------------------------------|-------------------------------------------------|
+| galleryAttachmentInjector | Single or multiple images and video attachments | `_GalleryAttachmentViewInjector<ExtraData>.self`|
+| linkAttachmentInjector    | URL preview attachments                         | `_LinkAttachmentViewInjector<ExtraData> .self`  |
+| giphyAttachmentInjector   | Giphy attachments                               | `_GiphyAttachmentViewInjector<ExtraData> .self` |
+| filesAttachmentInjector   | File attachments                                | `_FilesAttachmentViewInjector<ExtraData> .self` |
+
+You can implement `MyCustomAttachmentViewInjector` as a subclass of `FilesAttachmentViewInjector` or as a subclass of `AttachmentViewInjector`.
+
+In both cases you will implement at least these two methods: `contentViewDidLayout(options: ChatMessageLayoutOptions)` and `contentViewDidUpdateContent`. 
+
+To keep this easy to read we are going to create two classes: `MyCustomAttachmentViewInjector` and `MyCustomAttachmentView`. The latter can be a simple View class
+or you can also use StreamChatUI's [_View](../ui-components/CommonViews/_View) class which is what we recommended.
+
+```swift
```

Comment:
If this code block is just boilerplate as a starting point, it is fine.

If not, it's a good idea to add 1-2 senteces in code comments above the blocks in this to explain why each block is there.